src/battle_script_commands.c

```diff
@@ -1895,15 +1895,52 @@ static void Cmd_ppreduce(void)
 }
 
 // The chance is 1/N for each stage.
-#if B_CRIT_CHANCE >= GEN_7
-    static const u8 sCriticalHitOdds[] = {24, 8, 2, 1, 1};
-#elif B_CRIT_CHANCE == GEN_6
-    static const u8 sCriticalHitOdds[] = {16, 8, 2, 1, 1};
-#else
-    static const u8 sCriticalHitOdds[] = {16, 8, 4, 3, 2}; // Gens 2,3,4,5
-#endif // B_CRIT_CHANCE
+static const u32 sGen7CriticalHitOdds[] = {24, 8, 2, 1, 1};
+static const u32 sGen6CriticalHitOdds[] = {16, 8, 2, 1, 1};
+static const u32 sCriticalHitOdds[]     = {16, 8, 4, 3, 2}; // Gens 2,3,4,5
+
+static inline u32 GetCriticalHitOdds(u32 critChance)
+{
+    if (B_CRIT_CHANCE >= GEN_7)
+        return sGen7CriticalHitOdds[critChance];
+    if (B_CRIT_CHANCE == GEN_6)
+        return sGen6CriticalHitOdds[critChance];
+
+    return sCriticalHitOdds[critChance];
+}
+
+#define GEN_1_FARFETCHED_LEEK_SCALER 8
```

Comment:
If we want to handle Gen 1 Leek with this function, I think we should also handle Gen 1 Lucky Punch and Scope Lens, have similar defines for them, and use this function in the Gen 1 version also.

_Or_ we don't use this function in the gen 1 version at all, if you don't want to have 3 defines here. I don't like the idea of using this function for some gen 1 held items but not all of them.

Reply:
fair point! will work out a solution tomorrow